### PR TITLE
Added the ability to customise the script output directory

### PR DESCRIPTION
**Pull request by codemonkey76**
Add the ability to customise the script output directory by adding an option to gleam.toml
```toml
[build]
target-dir="bin"
```

### CODE REVIEW

File: src/gleescript.gleam [hunk 1]
```diff
@@ -51,17 +51,34 @@ pub fn run() -> snag.Result(Nil) {
...
+      simplifile.create_directory_all(config.target_dir)
+      |> snag_inspect_error
+      |> snag.context("Failed to create " <> config.target_dir <> " directory")
+    }
+    Error(e) ->
+      Error(e)
+      |> snag_inspect_error
+      |> snag.context("Failed to read " <> config.target_dir <> " directory")
+    Ok(True) -> Ok(Nil)
+  })
```
Comment (lpil) [inline src/gleescript.gleam:70]: We can call `create_directory_all` without calling `verify_is_directory` first here.

File: src/gleescript.gleam [hunk 2]
```diff
@@ -121,7 +142,12 @@ fn load_config() -> snag.Result(Config) {
     |> snag.context("Failed to get package name from gleam.toml"),
   )
 
-  Ok(Config(package_name: package_name))
+  let target_dir = case tom.get_string(config, ["build", "target-dir"]) {
+    Ok(target) -> "./" <> target <> "/"
+    Error(_) -> "./"
+  }
```
Comment (lpil) [inline src/gleescript.gleam:148]: We'll need to support absolute directories and trailing slashes here.

**Review by lpil — COMMENTED** [review]
Thank you. Let's use a command line flag rather than adding to `gleam.toml`. I don't want to use any parser library, rather can iterate over a list of arguments from the `argv` package to determine the given flags.

File: src/gleescript.gleam (hunk 1 above)
Comment (codemonkey76) [reply src/gleescript.gleam:70]: noted

File: src/gleescript.gleam (hunk 2 above)
Comment (codemonkey76) [reply src/gleescript.gleam:148]: noted.

**Comment by codemonkey76** [comment]
I'm not sure how to test this.

**Review by lpil — COMMENTED** [review]
Fab!! Would you mind adding the test to CI? Thanks

File: test.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+DIRECTORY=test-dir
+
+if [ -d $DIRECTORY ]; then
+    echo "Directory $DIRECTORY already exists"
+    exit 1
+fi
```
[the remark this answers is not in the document]
Comment (codemonkey76) [reply test.sh]: done

**Review by lpil — APPROVED** [review]
Thank you

**Pull request merged**